### DISCUSSION TITLE
set a timer to workflow

### DISCUSSION THREAD

**Discussion by MahnoorK1** [bonsai-rx/bonsai#1023] [Q&A] [answered]
Hello, 

I need help on how to add a timer to my workflow such that the video capture will stop after the assigned duration and I only have to hit Start. I see two 'timer' nodes when I search but I don't know which one to use. I've tried them both and put in 10 seconds as an example but nothing happened and the recording did not stop after 10 seconds. 

Here is my workflow so far
![Bonsai screenshot](https://user-images.githubusercontent.com/112672417/188336482-a4a88339-c87f-4051-a3ef-d443a8548b18.jpeg)

I have two webcams that I'm getting video out of, and I want the timer to apply to both of them at the same time.

Appreciate any help, thanks in advance!

**Comment by bruno-f-cruz** [comment, marked as answer]
Heya,

If you want a timer that is specific for the sequences of the cameras you can use `TakeUntil` with a `Timer` as the triggering event:
![image](https://user-images.githubusercontent.com/7049351/188394805-e3cd9e45-6fea-4fb9-82fa-258e9b64d29c.png)

Alternatively, if you want to stop the whole workflow, as your title suggests, you can use a `Timer` connected to a `WorkflowOutput`. This last trick works because upon the termination of any sequence connected to a `WorkflowOutput`, the workflow will be closed.

![image](https://user-images.githubusercontent.com/7049351/188395165-766c72ef-432d-47df-92a4-ce75cf36abad.png)

Hope it helps,
Bruno
